[DOCS] Security ON by default alpha1 (#76052)
elastic/elasticsearch: x-pack/docs/en/security/configuring-stack-security.asciidoc

[[configuring-stack-security]]
== Start the Elastic Stack with security enabled

beta::[This functionality is in alpha and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Alpha features are not subject to the support SLA of official GA features.]

Before you start {es} for the first time, run the `elasticsearch-security-config`
tool. This tool generates certificates and keys for the transport and HTTP
layers of your, and writes the TLS configuration settings in `elasticsearch.yml`.

[discrete]
=== Prerequisites

Download the `elasticsearch 8.0.0-alpha1` package distribution for your
environment.

[discrete]
=== Start {es} with security enabled

> . Before starting {es} for the first time, run the

Reviewer: Would it make sense for us to add a paragraph here saying that this is going to be taken care automatically in the future the first time you start your node and that this is early preview showcasing the direction we're taking ? Or is this implied by it being alpha1 ?

Author: I feel like it's implied since this is an alpha release. IMO, the docs should reflect the current state of the release. We can create a new PR for alpha2 and update the docs to match the workflow and capabilities.

`elasticsearch-security-config` tool from the directory where you downloaded
the {es} package distribution:
+
[source,shell]
----
./bin/elasticsearch-security-config
----

> . Navigate to `$ES_PATH_CONF/auto_config_on_<timestamp>` to view the

Reviewer: Can we rewrite this to make it clear that they don't have to do this, but can if they want to view the keystores ?

Author: I'm going to combine this information into the previous step and indicate that the `elasticsearch-security-config` tool generates the following security certificates and keys in `$ES_PATH_CONF/auto_config_on_<timestamp>`.

Keystore for the first {es} node in your cluster.

`transport_keystore_all_nodes.p12`::
Keystore for adding additional nodes to your cluster.
--
+
NOTE: The `ES_PATH_CONF` variable is the path for the {es}
configuration files. If you installed {es} using archive distributions
(`zip` or `tar.gz`), the variable defaults to `ES_HOME/config`. If you used
package distributions (Debian or RPM), the variable defaults to
`/etc/elasticsearch`.

. Start {es}.
+
[source,shell]
----
./bin/elasticsearch
----

. In another terminal window, run the
<<setup-passwords,`elasticsearch-setup-passwords`>> utility from the directory
where you downloaded the {es} package distribution:
+
[source,shell]
----
./bin/elasticsearch-setup-passwords auto
----
+
If you want to use your own passwords, run the command with the
`interactive` parameter instead of the `auto` parameter. Using this mode
steps you through password configuration for all of the built-in users.
+
[source,shell]
----
./bin/elasticsearch-setup-passwords interactive
----

. Save the generated passwords. You'll need them to add the built-in user to
{kib}.
+
WARNING: After you set a password for the `elastic` user, you cannot run the

> `elasticsearch-setup-passwords` command a second time.

Reviewer: Add a note that if they want to change the password of the elastic user or if they lost it or they didn't write it down, they can always use `bin/elastisearch-reset-elastic-password`

Author: I'll change this WARNING to a NOTE, and then indicate how to change the password for the `elastic` user with the `elastisearch-reset-elastic-password` tool.

. Make an authenticated call to your {es} cluster and enter the password for
the `elastic` user when prompted:
+
[source,shell]
----

> curl -k -uelastic https://localhost:9200

Reviewer: ```suggestion
curl --cacert $ES_PATH_CONF/auto_config_on_<timestamp>/http_ca.crt -uelastic https://localhost:9200
```

Author: +1 -- I'll add that parameter to the `curl` command and provide some context about it.

[[configuring-stack-security]]
== Start the Elastic Stack with security enabled

beta::[This functionality is in alpha and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Alpha features are not subject to the support SLA of official GA features.]

Before you start {es} for the first time, run the `elasticsearch-security-config`
tool. This tool generates certificates and keys for the transport and HTTP

> layers of your, and writes the TLS configuration settings to `elasticsearch.yml`.

Reviewer: ```suggestion
layers of {es}, and writes the TLS configuration settings to `elasticsearch.yml`.
```

[discrete]
=== Prerequisites

Download the `elasticsearch 8.0.0-alpha1` package distribution for your
environment.

[discrete]
=== Start {es} with security enabled

. Before starting {es} for the first time, run the
`elasticsearch-security-config` tool from the directory where you downloaded
the {es} package distribution:
+
[source,shell]
----
./bin/elasticsearch-security-config
----
+
The `elasticsearch-security-config` tool generates the following security
certificates and keys in `$ES_PATH_CONF/auto_config_on_<timestamp>`:
+
--

> `http.p12`::

Reviewer: there is no such file

Reviewer: There is `http_ca.crt` which we can describe as : "The CA certificate that is used to sign the certificates for the HTTP layer of this cluster."

Author: Thanks for catching this copy/paste error. I'll update to `http.ca.crt` 👍

Certificate for the HTTP layer, which you can use to enroll additional clients
such as {kib}.

`http_keystore_local_node.p12`::

> Keystore for the first {es} node in your cluster.

Reviewer: ```suggestion
Keystore that contains the key and certificate for the HTTP layer for the this node.
```

`transport_keystore_all_nodes.p12`::

> Keystore for adding additional nodes to your cluster.

Reviewer: ```suggestion
Keystore that contains the key and certificate for the transport layer for all the nodes in your cluster.
```

of the `kibana_system` user.

This account is not meant for individual users and does not have permission to
log in to {kib} from a browser. Instead, you'll log in to {kib} as the `elastic`
superuser. 

. Add the `elasticsearch.username` setting to the `KIB_PATH_CONF/kibana.yml`
file and set the value to the `kibana_system` user:
+
[source,yaml]
----
elasticsearch.username: "kibana_system"
----
+
NOTE: The `KIB_PATH_CONF` variable is the path for the {kib}
configuration files. If you installed {kib} using archive distributions
(`zip` or `tar.gz`), the variable defaults to `KIB_HOME/config`. If you used
package distributions (Debian or RPM), the variable defaults to `/etc/kibana`.

. From the directory where you installed {kib}, run the following commands
to create the {kib} keystore and add the secure settings:

   a. Create the {kib} keystore:
+
[source,shell]
----
./bin/kibana-keystore create
----

   b. Add the password for the `kibana_system` user to the {kib} keystore:
+
[source,shell]
----
./bin/kibana-keystore add elasticsearch.password
----
+
When prompted, enter the password for the `kibana_system` user.

> 

Reviewer: Ping to add the relevant TLS config as discussed in slack

Author: Added the steps to configure TLS for Kibana in https://github.com/elastic/elasticsearch/pull/76052/commits/1a2da8d38bd7e5a28af78f409b25c732b43579ae

beta::[This functionality is in alpha and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Alpha features are not subject to the support SLA of official GA features.]

Before you start {es} for the first time, run the `elasticsearch-security-config`
tool. This tool generates certificates and keys for the transport and HTTP
layers of {es}, and writes the TLS configuration settings to `elasticsearch.yml`.

[discrete]
=== Prerequisites

Download the `elasticsearch 8.0.0-alpha1` package distribution for your
environment.

[discrete]
=== Start {es} with security enabled

. Before starting {es} for the first time, run the
`elasticsearch-security-config` tool from the directory where you downloaded
the {es} package distribution:
+
[source,shell]
----
./bin/elasticsearch-security-config
----
+
The `elasticsearch-security-config` tool generates the following security
certificates and keys in `$ES_PATH_CONF/auto_config_on_<timestamp>`:
+
--
`http_ca.crt`::
The CA certificate that is used to sign the certificates for the HTTP layer of
this {es} cluster.

`http_keystore_local_node.p12`::

> Keystore that contains the key and certificate for the HTTP layer for the this node.

Reviewer: ```suggestion
Keystore that contains the key and certificate for the HTTP layer for this node.
```
